Meta/Lagom/Tools/CodeGenerators/JSSpecCompiler/CompilationPipeline.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2023, Dan Klishch <danilklishch@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Function.h>
+#include <AK/StringView.h>
+
+#include "Forward.h"
+
+namespace JSSpecCompiler {
+
+class CompilationStep {
+public:
+    CompilationStep(StringView name)
+        : m_name(name)
+    {
+    }
+
+    virtual ~CompilationStep() = default;
+    virtual void run(TranslationUnitRef translation_unit) = 0;
+
+    StringView m_name;
```

Comment:
public member vars in `class` types are sad. Should this be a struct? or can we encapsulate the class's data members.